src/utf32.jl

```diff
@@ -137,10 +137,10 @@ end
 convert(::Type{UTF32String}, data::AbstractVector{Int32}) =
     convert(UTF32String, reinterpret(UInt32, convert(Vector{Int32}, data)))
 
-convert(::Type{UTF32String}, data::AbstractVector{Char}) =
+convert(::Type{UTF32String}, data::AbstractVector{<:AbstractChar}) =
     convert(UTF32String, map(UInt32, data))
 
-convert(::Type{T}, v::AbstractVector{S}) where {T<:AbstractString, S<:Union{UInt32,Char,Int32}} =
+convert(::Type{T}, v::AbstractVector{S}) where {T<:Union{ASCIIString,UTF16String}, S<:Union{UInt32,AbstractChar,Int32}} =
```

Comment:
Note that the old method here was type piracy.  (There are probably some other instances of type piracy in this ancient package, but I don't have the inclination to conduct an audit.)